core/include/scipp/core/variable_view.h

```diff
@@ -148,6 +148,13 @@ template <class T> class SCIPP_CORE_EXPORT VariableView {
     return std::equal(begin(), end(), other.begin());
   }
 
+  /// Return true if *this and other are two equivalent views of the same data.
+  bool isSame(const VariableView<T> &other) const {
+    return (m_variable == other.m_variable) && (m_offset == other.m_offset) &&
+           (m_dimensions == other.m_dimensions) &&
+           (m_targetDimensions == other.m_targetDimensions);
+  }
```

Comment:
As far as I can tell, no direct testing of newly added methods `isSame` though other indirect testing looks good.